tests/codegen/patchable-function-entry.rs

```diff
@@ -0,0 +1,28 @@
+//@ compile-flags: -Z patchable-function-entry=15,10
```

Comment:
Please create additional tests, or [additional _revisions_ of this test](https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#revisions), so that we thoroughly test the following cases, and have documentation for the intended behavior of:
- [x] when `-Zpatchable-function-entry` is set but the function has an attribute
- [x] when `-Zpatchable-function-entry` is set but these functions are not given attributes
- [ ] when there are `#[patchable_function_entry] fn`s but `-Zpatchable-function-entry` is NOT set
- [ ] `-Zpatchable-function-entry={t}` has `t` < a `#[patchable_function_entry] fn`s `prefix_nops + entry_nops`
- [ ] `-Zpatchable_function-entry={t},{e}` has `e` < a `#[patchable_function_entry] fn`s `entry_nops`

I realize many of these are "obvious" but it helps to have covered these cases to prevent them regressing later because someone was careless.

Reply:
Should be done, I think. I added a bunch of additional tests that made sense to me, while I doing this.